kernel/init.c

```diff
@@ -764,6 +771,9 @@ FUNC_NORETURN void z_cstart(void)
 #endif /* CONFIG_MULTITHREADING */
 	/* do any necessary initialization of static devices */
 	z_device_state_init();
+#if CONFIG_ARCH_CACHE
+	arch_cache_init();
+#endif
```

Comment:
As we only need it for ARC - we can also move it to `z_prep_c` and avoid adding empty functions for all other architectures. We can place it before `arc_mpu_init`.

In rest - LGTM.

Reply:
`z_prep_c `would work, but as (internal) cache initialization is a generic operation and we have the `sys_cache_*` generic functions as well, a generic init makes sense to me. And caches are more part of architecture than C runtime initialization.
Benefit of `z_prep_c` would be caches can be enabled/used sooner. Any other place this init could be moved to, like `arch_kernel_init`?